### DISCUSSION TITLE
Netplay - Pokemon

### DISCUSSION THREAD

**Discussion by Jmallone** [OnionUI/Onion#1109] [Ideas]
Hello, this feature of Easy Netplay - Pokemon is wonderful

 , I would like to know if there is an implementation perspective for playing Online outside the local network?

**Comment by XK9274** [comment]
Thanks, glad you're enjoying it :)

Currently not a feature to play globally but it's on my to-do list to see if it's possible.. it's got a fair few hurdles.

**Reply by Jmallone** [reply]
Very cool, is there anything I can do about programming?  a "good first issue" for this problem?  


I saw in the code that a wifi hostspot is created, what do you think about putting servers like the ones in minecraft?  then people could add their own to a list and connect, but they would have to develop this application separately to be hosted on a computer.

**Reply by XK9274** [reply]
> Very cool, is there anything I can do about programming?  a "good first issue" for this problem?  
> 
> 
> I saw in the code that a wifi hostspot is created, what do you think about putting servers like the ones in minecraft?  then people could add their own to a list and connect, but they would have to develop this application separately to be hosted on a computer.

Yeah the scripts handle a lot in the backend, all of which needs to be done for global with the added complexity of somehow communicating freely like the scripts do which over the internet is a lot more difficult than within a LAN, especially when you don't want users to have to do anything with their FW (to keep things as 'Easy' as possible..) 

Like I say, it's not impossible.. it's just a bit more complex.

Also we're just gonna rely on the retroarch for servers and joining but one thing me and gecko discussed was the use of a relay server (which RA supports anyway) but the process isn't easy to work into retroarch as they generally don't handle much in terms of file transfer and free/open communication like we'd need. 

I've got some ideas, just not so much time rn 😁
